.github/workflows/e2e_lighthouse_performance_acceptance_tests.yml

```diff
@@ -125,64 +75,29 @@ jobs:
       - uses: ./.github/actions/merge
-      - name: Cache node modules
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: /home/runner/work/oppia/yarn_cache
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - name: Attempt to download build files artifact
-        id: download_artifact_1
-        uses: actions/download-artifact@v3
-        continue-on-error: true
-        with:
-          name: build_files
-          path: /home/runner/work/oppia/
-      - name: Unzip build files
-        if: steps.download_artifact_1.outcome != 'failure'
+      # Installing python dependencies becuase we are executing a python script for installing chrome in CI.
+      - name: Install Python deps
         run: |
-          ls -la
-          unzip build_files.zip
-          rm build_files.zip
-          echo "Files in ./:"
-          ls -la .
-          echo "Files in oppia_tools:"
-          ls -la oppia_tools
-          echo "Files in oppia:"
-          ls -la oppia
-          echo "Files in build:"
-          ls -la oppia/build
-          echo "Files in third_party:"
-          ls -la oppia/third_party
-        working-directory: /home/runner/work/oppia
-        shell: bash
-      - uses: ./.github/actions/install-oppia-dependencies
+          pip install --require-hashes --no-deps -r requirements.txt
+          pip install --require-hashes --no-deps -r requirements_dev.txt
       - name: Install chrome
         run: python -m scripts.install_chrome_for_ci
       - name: Install ffmpeg
         run: sudo apt install ffmpeg
-      - name: (Artifact re-build) Cache third_party/static
-        if: steps.download_artifact_1.outcome == 'failure'
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - name: (Artifact re-build) Build Webpack
-        if: steps.download_artifact_1.outcome == 'failure'
-        run: python -m scripts.build --prod_env
+      # Deleting ubuntu files for getting more space (refer: https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930)
+      - name: Deleting ubuntu files
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+      - name: Initializing Containers
+        run: |
+          sudo make build
+          sudo docker compose up angular-build -d
+          sudo docker cp oppia-angular-build:/app/oppia/node_modules .
```

Comment:
consider it as optimisation, we can do the same using make run-devserver command, but this command takes more time since it waits for the server to start completely. it is not required here since we again start the server with the make run_tests.e2e, so it is not starting the server here and saves us around 7mins on CI. please refer [here](https://github.com/oppia/oppia/pull/18999#discussion_r1377376759) for why we need to strt the angular build container at first

Reply:
> Please leave a code-comment

Ah, I was thinking of writing code-comment i.e, within the file explaining why we need to copy node module. I think it's because currently we are running e2e through host, right? ALso, add TODO(..) if we are planning to ffix it in ffuture.

Reply:
> ALso, add TODO(..) if we are planning to ffix it in ffuture.

are you asking this as we execute e2e, acceptance and lighthouse test on host? if yes, do we need them to execute within the containers only and in the long run would it be beneficial to execute these tests in the containers or host?

Reply:
added in issue https://github.com/oppia/oppia/issues/19134. also, added a todo above the step name